addons/controllers/clusterbootstrap_controller.go

```diff
@@ -1087,6 +1097,50 @@ func (r *ClusterBootstrapReconciler) periodicGVRCachesClean() {
 	}
 }
 
+// createSecretFromInline creates a Secret from inline config in valuesFrom
+func (r *ClusterBootstrapReconciler) createSecretFromInline(cluster *clusterapiv1beta1.Cluster,
+	pkg *runtanzuv1alpha3.ClusterBootstrapPackage, pkgRefName string, log logr.Logger) (*corev1.Secret, error) {
+
+	inlineSecret := &corev1.Secret{}
+	inlineSecret.Name = util.GeneratePackageSecretName(cluster.Name, pkgRefName)
+	// The secret will be created or patched under tkg-system namespace on remote cluster
+	inlineSecret.Namespace = cluster.Namespace
+	inlineSecret.Type = corev1.SecretTypeOpaque
+
+	var createOrPatchErr error
+	_, createOrPatchErr = controllerutil.CreateOrPatch(r.context, r.Client, inlineSecret, func() error {
+		inlineSecret.OwnerReferences = []metav1.OwnerReference{
+			{
+				APIVersion: clusterapiv1beta1.GroupVersion.String(),
+				Kind:       cluster.Kind,
+				Name:       cluster.Name,
+				UID:        cluster.UID,
+			},
+		}
+
+		inlineSecret.Data = map[string][]byte{}
+		inlineConfigYamlBytes, err := yaml.Marshal(pkg.ValuesFrom.Inline)
```

Comment:
question: Since we are changing the pkg.ValuesFrom.Inline to be a map[string]interface{}, wondering how does it handle the YTT headers like:
```
#@data/values
#@overlay/match-child-defaults missing_ok=True
```
If we need to consider the YTT header case, do we want to prepend it to `inlineConfigYamlBytes` ?